src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1846,6 +1851,21 @@ private static void ReportImplicitImplementationMismatchDiagnostics(Symbol inter
             }
         }
 
+        internal static bool MethodsHaveSameInitOnly(Symbol one, Symbol other)
+        {
+            if (!(one is MethodSymbol oneMethod))
+            {
+                return true;
```

Comment:
>return true; [](start = 16, length = 12)

This is very confusing. Why would we return true when this isn't even a method. Consider renaming method to return true for mismatch and returning true when both are methods and there is a mismatch